pkg/tests/regress_1020_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	testutils "github.com/pulumi/pulumi-terraform-bridge/testing/x"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
+	"net"
+)
+
+// See https://github.com/pulumi/pulumi-terraform-bridge/issues/1020
+func TestRegress1020(t *testing.T) {
+	ctx := context.Background()
+
+	CIDRBlocksEqual := func(cidr1, cidr2 string) bool {
+		ip1, ipnet1, err := net.ParseCIDR(cidr1)
+		if err != nil {
+			return false
+		}
+		ip2, ipnet2, err := net.ParseCIDR(cidr2)
+		if err != nil {
+			return false
+		}
+
+		return ip2.String() == ip1.String() && ipnet2.String() == ipnet1.String()
+	}
+
+	emptySchema := func() *schema.Schema {
+		return &schema.Schema{
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{},
+			},
+		}
+	}
+
+	resource := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 10000,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if d.GetRawPlan().IsNull() {
+						panic(fmt.Sprintf("%s", "NULL GetRawPlan"))
```

Comment:
d.GetRawPlan() is null but should not be, therefore d.GetRawPlan().GetAttr("addresses") panics.

Digging further.

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/issues/1021 is probably related.